mlrun/runtimes/sparkjob.py

```diff
@@ -170,20 +172,44 @@ class SparkRuntime(KubejobRuntime):
     apiVersion = group + "/" + version
     kind = "spark"
     plural = "sparkapplications"
-
-    @property
-    def _default_image(self):
+    default_mlrun_image = ".spark-job-default-image"
+    gpu_suffix = "-cuda"
+
+    @classmethod
+    def _default_mlrun_image_name(cls, with_gpu=False):
+        suffix = cls.gpu_suffix if with_gpu else ""
+        return cls.default_mlrun_image + suffix
+
+    @classmethod
+    def deploy_default_image(cls, with_gpu=False):
+        from mlrun.run import new_function
+
+        sj = new_function(kind=cls.kind, name="spark-default-image-deploy-temp")
+        sj.spec.build.image = cls._default_mlrun_image_name(with_gpu)
+        if with_gpu:
```

Comment:
I guess that you could shorten this like this, no?
sj.with_executor_requests(cpu=1, mem="512m", gpus=1 if with_gpu else 0)